habitat-lab/habitat/tasks/rearrange/actions/oracle_nav_action.py

```diff
@@ -282,6 +282,155 @@ def step(self, *args, is_last_action, **kwargs):
                 )
 
 
+@registry.register_task_action
```

Comment:
Again, since this is core Habitat-lab functionality, it should be a separate PR, into `SIRo_dev`.

I will hold off reviewing this part and instead let other folks review this part when the new PR goes up.